netlink.go

```diff
@@ -67,26 +68,32 @@ func Loopback(ctx context.Context, d Device, size uint64) (idx uint32, wait func
 		return 0, nil, err
 	}
 
+	configErr := make(chan error, 1)
+	defer close(configErr)
+
 	ctx, cancel := context.WithCancel(ctx)
 	ch := make(chan error, 1)
-	go func() {
-		<-ctx.Done()
-		client.Close()
-	}()
 	go func() {
 		err := serve(ctx, serverc, connParameters{exp, defaultBlockSizes})
 		if e := ctx.Err(); e != nil {
 			err = e
 		}
 		cancel()
-		ch <- err
+		client.Close()
```

Comment:
I moved the `client.Close()` here to ensure that `wait()` waits for it to happen. Otherwise I think this could've potentially been called after the process closed, though it doesn't really matter since the OS would've freed it anyway. Either way, it seemed simpler to just have one goroutine that does all the cleanup?